### ISSUE TITLE
False `UndefinedMagicMethod`: custom builder subclass methods lost through a fluent chain (`#[UseEloquentBuilder]`)

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1216]
## Description

Methods declared on a **custom Eloquent builder subclass** are reported as false `UndefinedMagicMethod` when they are reached through a fluent chain whose intermediate type widens to the abstract parent (or base `Builder`), even though they resolve fine at runtime.

The model opts into the custom builder via `#[UseEloquentBuilder(...)]` (Laravel 12+), so this is *not* the `@mixin \...\Model` case in #1215 — no `@mixin` is involved here.

## Reproduction

Found auditing [koel/koel](https://github.com/koel/koel) (Laravel 13). Minimal shape of `app/Builders/`:

```php
abstract class FavoriteableBuilder extends \Illuminate\Database\Eloquent\Builder
{
    public function accessible(): static { /* ... */ return $this; }
}

class ArtistBuilder extends FavoriteableBuilder
{
    public function withUserContext(User $user): static
    {
        return $this
            ->accessible()
            ->when(true, static fn (self $q) => $q)   // Conditionable::when()
            ->withRatingSubquery();                    // <-- false UndefinedMagicMethod
    }

    private function withRatingSubquery(): self { /* ... */ return $this; }
}
```

```php
#[\Illuminate\Database\Eloquent\Attributes\UseEloquentBuilder(ArtistBuilder::class)]
class Artist extends Model { /* ... */ }
```

## Actual behavior

```
UndefinedMagicMethod: Magic method App\Builders\FavoriteableBuilder::withRatingSubquery does not exist
  at app/Builders/ArtistBuilder.php:141
```

Note the receiver resolves to the **abstract parent** `FavoriteableBuilder`, not the concrete `ArtistBuilder` — the chain through `accessible()` / `when()` lost the late-static-bound subclass, so the child's method is not found.

## Expected behavior

The chain should preserve the concrete builder subclass (`static`), so `withRatingSubquery()` resolves on `ArtistBuilder`. No `UndefinedMagicMethod`.

## Notes / likely cause

- The intermediate methods (`accessible()` returns `static`; Laravel's `Conditionable::when()` returns `$this`) collapse to the parent/base builder type in the plugin's builder handling, dropping the subclass. Related closed work: #630 (direct `query()->scope()`), #901 (abstract base). This is the **chained** variant on a `#[UseEloquentBuilder]` subclass.
- Real-world hits in koel: `ArtistBuilder`/`AlbumBuilder`/`SongBuilder::withRatingSubquery()` and `withLengthSubquery()`.
